src/Avalanche/Address.h

```diff
@@ -0,0 +1,66 @@
+// Copyright © 2017-2021 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#pragma once
+
+#include "../Bech32Address.h"
+#include "../Data.h"
+#include "../PublicKey.h"
+
+#include <string>
+
+namespace TW::Avalanche {
+
+/// Avalanche address is a Bech32Address, with "avax" prefix and Sha2 hash; and then the chain prefixed to that.
+class Address : public Bech32Address {
+private:
+    void extractKeyHashFromString (const std::string& string);
+public:
+    static const size_t hashLen;
+    static const std::string hrp;
+
+    static bool isValid(const std::string& addr);
+
+    Address() : Bech32Address(hrp) {}
+
+    /// Initializes an address with a key hash.
+    Address(const Data& keyHash) : Bech32Address(hrp, keyHash) {}
+
+    /// Initializes an address with a public key.
+    Address(const PublicKey& publicKey) : Bech32Address(hrp, HASHER_SHA2_RIPEMD, publicKey) {}
+
+    /// Initializes an address from a string.
+    Address(const std::string& string) : Bech32Address(hrp) {
+        extractKeyHashFromString(string);
+    }
+
+    static bool decode(const std::string& addr, Address& obj_out) {
+        return Bech32Address::decode(addr, obj_out, hrp);
+    }
+
+    std::string string() const;
+};
+
+inline bool operator==(const Address& lhs, const Address& rhs) {
+    // prefer string over byte comparison because string encodes chain and net
+    return lhs.string() == rhs.string();
+}
+
+inline bool operator!=(const Address& lhs, const Address& rhs) {
+    return lhs.string() != rhs.string();
+}
+
+inline bool operator<(const Address& lhs, const Address& rhs) {
```

Comment:
is this operator override needed?

Reply:
I think so, it's important that any `std::sort`ing of vectors of `Address`es do a lexicographical sort on just the keyhash bytes. Unless this is redundant and I don't see it?